upstairs/src/downstairs.rs

```diff
@@ -474,12 +474,77 @@ impl Downstairs {
         }
     }
 
+    /// Checks whether an ack and/or a retire check is necessary for the job
```

Comment:
This is exactly the follow up to 1552 I was going to chat to you about, but it was next in the queue :) :+1: